src/validation.cpp

```diff
@@ -3681,6 +3694,51 @@ bool CVerifyDB::VerifyDB(const CChainParams& chainparams, CCoinsView *coinsview,
     return true;
 }
 
+bool ReplayBlocks(const CChainParams& params, CCoinsView* view)
+{
+    LOCK(cs_main);
+
+    CCoinsViewCache cache(view);
+
+    uint256 hashBest = view->GetBestBlock();
+    uint256 hashUpto = view->GetUptoBlock();
+
+    if (hashUpto.IsNull() || hashBest == hashUpto) return true;
+
+    if (mapBlockIndex.count(hashUpto) == 0) {
+        return error("ReplayBlocks(): chainstate boundaries not in block index");
+    }
+    auto pindexUpto = mapBlockIndex[hashUpto];
+
+    int nHeight = 1; // Skip the genesis block
+    if (mapBlockIndex.count(hashBest)) {
```

Comment:
Should we also fail here if hashBest has been written (ie is non-IsNull) but isnt in mapBlockIndex?